freud/locality.pyx

```diff
@@ -181,6 +280,37 @@ cdef class NeighborQuery:
     def points(self):
         return np.asarray(self.points)
 
+    def _queryGeneric(self, points, query_args):
+        # This function is temporarily included for testing and WILL be
+        # removed in future releases.
```

Comment:
Can you help me understand what your plan is for this? Should it _ever_ be released if you're planning to remove it soon?

Reply:
We can't test this functionality without exposing it to Python. My original thought was that we would remove it once we refactor the rest of freud to use NeighborQuery objects, since those would implicitly test the function. On the other hand, we could just leave this as a private function that exists only for unit tests. Now that I've written so many tests, I would be open to the latter. Thoughts?

Reply:
I don't see a reason why we should remove it. It might even be useful in the public API someday. I would take off this comment about it being removed in the future.

Reply:
After discussing offline, we clarified that enabling a public API with this method requires adding substantial logic to the Cython side that is currently all in C++. As a result, we will not plan to implement this now, but may do so in the future.